File: haskell/iniciante/1011.hs
-- title: Esfera
-- number of the problem: 1011
-- date: 20/09/20
-- author: ltsdw

import Text.Printf

myPI :: Double
myPI = 3.14159

volume :: Double -> Double
volume radius = (4.0 / 3.0) * myPI * (radius ** 3)

printVolume :: Double -> IO ()
printVolume volume = do
    printf "VOLUME = %.3f\n" volume


main :: IO ()
main = do
    radius' <- getLine

    let radius = read radius' :: Double

    let vol = volume radius

    printVolume vol

File: c++/iniciante/src/1011.cpp
/* title: Esfera
* number of the problem: 1011
* date: 30/08/21
* author: ltsdw */

#include <iostream>
#include <cmath>
#include <iomanip>

double volume(const double& radius)
{
    using std::pow;

    constexpr double PI{3.14159};

    return 4 * PI * pow(radius, 3) / 3;
}

int main(int argc, const char* argv[])
{
    using std::cin, std::cout, std::fixed, std::setprecision;

    double radius;
    cin >> radius;

    cout << fixed << setprecision(3);
    cout << "VOLUME = " << volume(radius) << '\n';

    return EXIT_SUCCESS;
}
